iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/MultiplexingClientTests.java

```diff
@@ -1397,6 +1399,107 @@ public void registerDevicesWithIncorrectCredentialsBeforeOpenThrowsOnOpen() thro
+        Device nonMultiplexedDevice = Device.createDevice(UUID.randomUUID().toString(), AuthenticationType.SAS);
+        registryManager.addDevice(nonMultiplexedDevice);
+        String deviceConnectionString = registryManager.getDeviceConnectionString(nonMultiplexedDevice);
+
+        // intentionally change the hostname of the device to simulate registering a device with a different hostname
+        // to a multiplexing client. It shouldn't matter that the hostname itself isn't tied to an actual IoT Hub since
+        // no network requests should be made before this hostname validation.
+
+        String actualHostName = IotHubConnectionString.createConnectionString(iotHubConnectionString).getHostName();
+        deviceConnectionString = deviceConnectionString.replace(actualHostName, "some-fake-host-name.azure-devices.net");
+
+        DeviceClient deviceClientWithDifferentHostName = new DeviceClient(deviceConnectionString, testInstance.protocol);
+
+        registrationsUnwindForUnsupportedOperationExceptions(deviceClientWithDifferentHostName);
+    }
+
+    @Test
+    public void registrationsUnwindForDifferentProtocolClient() throws Exception
+    {
+        Device newDevice = Device.createDevice(UUID.randomUUID().toString(), AuthenticationType.SAS);
+        registryManager.addDevice(newDevice);
+        String deviceConnectionString = registryManager.getDeviceConnectionString(newDevice);
+
+        // Protocol for the new client is AMQPS if the test parameters are for AMQPS_WS, and vice versa. MultiplexingClient
+        // should throw an exception since this new client's protocol doesn't match, even though both AMQPS and AMQPS_WS are valid
+        // protocols
+        IotHubClientProtocol protocol = testInstance.protocol == IotHubClientProtocol.AMQPS ? IotHubClientProtocol.AMQPS_WS : IotHubClientProtocol.AMQPS;
+
+        DeviceClient differentProtocolDeviceClient = new DeviceClient(deviceConnectionString, protocol);
+        registrationsUnwindForUnsupportedOperationExceptions(differentProtocolDeviceClient);
+    }
+
+    public void registrationsUnwindForUnsupportedOperationExceptions(DeviceClient unsupportedDeviceClient) throws Exception
+    {
+        testInstance.setup(DEVICE_MULTIPLEX_COUNT);
+        this.testInstance.multiplexingClient.unregisterDeviceClients(this.testInstance.deviceClientArray);
+
+        this.testInstance.deviceClientArray.add(unsupportedDeviceClient);
+
+        boolean expectedExceptionThrown = false;
+        try
+        {
+            this.testInstance.multiplexingClient.registerDeviceClients(this.testInstance.deviceClientArray);
+        }
+        catch (UnsupportedOperationException e)
+        {
+            expectedExceptionThrown = true;
+        }
+
+        assertTrue("Expected bulk registration to throw an UnsupportedOperationException", expectedExceptionThrown);
+
+        // Because one of the devices in the bulk registration was MQTT, and therefore unsupported, none of the other clients
```

Comment:
nit: you probably forgot to update this - I see you're calling this method from other test scenarios as well (x509 device, device already opened etc.), so the desc shouldn't say "mqtt" specifically.